eva-accession-release/src/main/java/uk/ac/ebi/eva/accession/release/io/AccessionedVariantMongoReader.java

```diff
@@ -94,34 +96,37 @@ public void open(ExecutionContext executionContext) throws ItemStreamException {
     }
 
     @Override
-    public Variant read() throws Exception, UnexpectedInputException, ParseException, NonTransientResourceException {
-        return cursor.hasNext() ? getVariant(cursor.next()) : null;
+    public List<Variant> read() throws Exception, UnexpectedInputException, ParseException, NonTransientResourceException {
+        return cursor.hasNext() ? getVariants(cursor.next()) : null;
     }
 
-    Variant getVariant(Document clusteredVariant) {
+    List<Variant> getVariants(Document clusteredVariant) {
         String contig = clusteredVariant.getString(CONTIG_FIELD);
         long start = clusteredVariant.getLong(START_FIELD);
         long rs = clusteredVariant.getLong(ACCESSION_FIELD);
-        String reference = "";
-        String alternate = "";
-        long end = start;
-        List<VariantSourceEntry> sourceEntries = new ArrayList<>();
         String type = clusteredVariant.getString(TYPE_FIELD);
         String sequenceOntology = VariantTypeToSOAccessionMap.getSequenceOntologyAccession(VariantType.valueOf(type));
 
+        Map<String, Variant> variants = new HashMap<>();
         Collection<Document> submittedVariants = (Collection<Document>)clusteredVariant.get(SS_INFO_FIELD);
-        for (Document submitedVariant : submittedVariants) {
-            reference = submitedVariant.getString(REFERENCE_ALLELE_FIELD);
-            alternate = submitedVariant.getString(ALTERNATE_ALLELE_FIELD);
-            end = calculateEnd(reference, alternate, start);
-            String study = submitedVariant.getString(STUDY_FIELD);
-            sourceEntries.add(buildVariantSourceEntry(study, sequenceOntology));
+        for (Document submittedVariant : submittedVariants) {
+            String reference = submittedVariant.getString(REFERENCE_ALLELE_FIELD);
+            String alternate = submittedVariant.getString(ALTERNATE_ALLELE_FIELD);
+            long end = calculateEnd(reference, alternate, start);
+            String study = submittedVariant.getString(STUDY_FIELD);
+            VariantSourceEntry sourceEntry = buildVariantSourceEntry(study, sequenceOntology);
+
+            String variantId = contig + "_" + start + "_" + reference + "_" + alternate;
```

Comment:
Shall we do "(contig + "_" + start + "_" + reference + "_" + alternate).toUpperCase()"? We never know what brain-damaged surprise dbSNP data has in store for us 😄